### PR TITLE
Fix Zee5IE and also add Subtitles extraction.

### PR DESCRIPTION
**Pull request by Ashish0804**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Closes https://github.com/yt-dlp/yt-dlp/issues/728

### CODE REVIEW

**Comment by Ashish0804** [comment]
This should work for most of the free stuff.
But still have to implement using cookies and/or user:pass to get `trial` episodes in premium series.

**Comment by Ashish0804** [comment]
Should be more or less okay to merge

Btw imo `username, password = self._get_login_info()` should return `username.strip()` and `password.strip()` else `RETURN` is returned as password which is not `None`

File: yt_dlp/extractor/zee5.py [hunk 1]
```diff
@@ -83,16 +83,48 @@ class Zee5IE(InfoExtractor):
         'only_matching': True
     }]
     _DETAIL_API_URL = 'https://spapi.zee5.com/singlePlayback/getDetails?content_id={}&device_id=iIxsxYf40cqO3koIkwzKHZhnJzHN13zb&platform_name=desktop_web&country=IN&check_parental_control=false'
+    _OTP_REQUEST_URL = 'https://b2bapi.zee5.com/device/sendotp_v1.php?phoneno=91{}'
+    _OTP_VERIFY_URL = 'https://b2bapi.zee5.com/device/verifyotp_v1.php?phoneno=91{}&otp={}&guest_token=iIxsxYf40cqO3koIkwzKHZhnJzHN13zb&platform=web'
+    _USER_TOKEN = None
+    _LOGIN_HINT = 'Use "--username <mobile_number>" to login using otp or "--username token" and "--password <user_token>" to login using user token.'
+    _NETRC_MACHINE = 'zee5'
```
Comment (pukkandan) [inline yt_dlp/extractor/zee5.py]: too much stuff here. vars that is used only once don't need to be properties

File: yt_dlp/extractor/zee5.py [hunk 2]
```diff
@@ -83,16 +83,48 @@ class Zee5IE(InfoExtractor):
...
             'https://useraction.zee5.com/token/platform_tokens.php?platform_name=web_app',
             video_id, note='Downloading access token')
         data = {
-            "X-Z5-Guest-Token": "iIxsxYf40cqO3koIkwzKHZhnJzHN13zb",
             "x-access-token": access_token_request['token']
         }
+        if self._USER_TOKEN:
+            data['Authorization'] = 'bearer %s' % self._USER_TOKEN
+        else:
+            data['X-Z5-Guest-Token'] = 'iIxsxYf40cqO3koIkwzKHZhnJzHN13zb'
```
Comment (pukkandan) [inline yt_dlp/extractor/zee5.py:126]: repeated in 3 places. make this a property

File: yt_dlp/extractor/zee5.py [hunk 3]
```diff
@@ -80,53 +82,91 @@ class Zee5IE(InfoExtractor):
...
+    _LOGIN_HINT = 'Use "--username <mobile_number>" to login using otp or "--username token" and "--password <user_token>" to login using user token.'
+    _NETRC_MACHINE = 'zee5'
+
+    def _login(self):
+        username, password = self._get_login_info()
+        if username:
+            if len(username) == 10 and username.isdigit() and self._USER_TOKEN is None:
+                self.report_login()
+                otp_request_json = self._download_json('https://b2bapi.zee5.com/device/sendotp_v1.php?phoneno=91{}'.format(username),
+                                                       None, note="Sending OTP")
```
Comment (pukkandan) [inline yt_dlp/extractor/zee5.py:97]:
```suggestion
                                                       None, note='Sending OTP')
```

File: yt_dlp/extractor/zee5.py [hunk 4]
```diff
@@ -80,53 +82,91 @@ class Zee5IE(InfoExtractor):
...
+                if otp_request_json['code'] == 0:
+                    self.to_screen(otp_request_json['message'])
+                else:
+                    raise ExtractorError(otp_request_json['message'], expected=True)
+                otp_code = self._get_tfa_info('OTP')
+                otp_verify_json = self._download_json('https://b2bapi.zee5.com/device/verifyotp_v1.php?phoneno=91{}&otp={}&guest_token={}&platform=web'.format(username, otp_code, self._DEVICE_ID),
+                                                      None, note="Verifying OTP", fatal=False)
```
Comment (pukkandan) [inline yt_dlp/extractor/zee5.py:104]:
```suggestion
                                                      None, note='Verifying OTP', fatal=False)
```

File: yt_dlp/extractor/zee5.py [hunk 5]
```diff
@@ -80,53 +82,91 @@ class Zee5IE(InfoExtractor):
...
-            video_id, 'mp4', fatal=False)
-        mpd_url = try_get(
-            json_data,
-            (lambda x: x['video'][0], lambda x: x['video_details']['url']),
-            compat_str)
-        formats += self._extract_mpd_formats(
-            'https://zee5vod.akamaized.net' + mpd_url,
-            video_id, fatal=False)
+        data = {
+            "x-access-token": access_token_request['token']
```
Comment (pukkandan) [inline yt_dlp/extractor/zee5.py:122]:
```suggestion
            'x-access-token': access_token_request['token']
```

**Pull request merged**